src/js/components/help/downloads.js

```diff
@@ -0,0 +1,324 @@
+import { getCurrentUser, getDocsVersion, getIsEnterprise, getTenantCapabilities } from '../../selectors';
+import Time from '../common/time';
+import { getToken } from '../../auth';
+import { detectOsIdentifier } from '../../helpers';
+import Tracking from '../../tracking';
+
+// wget --auth-no-challenge --user "$HOSTED_MENDER_EMAIL" --password "$HOSTED_MENDER_PASSWORD" https://downloads.customer.mender.io/content/hosted/mender-monitor/debian/1.2.0/mender-monitor_1.2.0-1%2Bdebian%2Bbuster_all.deb
+
+const cookies = new Cookies();
+
+const osMap = {
+  MacOs: 'darwin',
+  Unix: 'linux',
+  Linux: 'linux'
+};
+
+const defaultArchitectures = ['armhf', 'arm64', 'amd64'];
+const defaultOSVersions = ['debian+buster', 'debian+bullseye', 'ubuntu+bionic', 'ubuntu+focal'];
+
+const getVersion = (versions, id) => versions[id] || 'master';
+
+const getToolLocation = ({ isPublic = true }) => (isPublic ? 'mender.io' : 'customer.mender.io/content/hosted');
+
+const defaultLocationFormatter = ({ os, tool, versionInfo }) => {
+  const location = getToolLocation(tool);
+  const { id, osList = [], title } = tool;
+  if (osList.length) {
+    return osList.reduce((accu, supportedOs) => {
+      const title = Object.entries(osMap).find(entry => entry[1] === supportedOs)[0];
+      accu.push({
+        location: `https://downloads.${location}/${id}/${getVersion(versionInfo, id)}/${supportedOs}/${id}`,
+        title,
+        isUserOs: osMap[os] === supportedOs
+      });
+      return accu;
+    }, []);
+  }
+  return [{ location: `https://downloads.${location}/${id}/${getVersion(versionInfo, id)}/linux/${id}`, title }];
+};
+
+const multiArchLocationFormatter = ({ tool, versionInfo }) => {
+  const location = getToolLocation(tool);
+  const { archList, id, osList, packageId } = tool;
+  return osList.reduce((accu, os) => {
+    const osArchitectureLocations = archList.map(arch => ({
+      location: `https://downloads.${location}/repos/debian/pool/main/m/${id}/${encodeURIComponent(
+        `${packageId || id}_${getVersion(versionInfo, id)}-1+${os}_${arch}.deb`
+      )}`,
+      title: `${os} - ${arch}`
+    }));
+    accu.push(...osArchitectureLocations);
+    return accu;
+  }, []);
+};
+
+const nonOsLocationFormatter = ({ tool, versionInfo }) => {
+  const location = getToolLocation(tool);
+  const { id, title } = tool;
+  return [
+    {
+      location: `https://downloads.${location}/${id}/${getVersion(versionInfo, id)}/${id}-${getVersion(versionInfo, id)}.tar.xz`,
+      title
+    }
+  ];
+};
+
+const defaultCurlDownload = ({ location, tokens }) => {
+  let headers = `-H cookie: JWT=${getToken()}`;
+  if (tokens.length) {
+    headers = `-H Authorization: Bearer ${tokens[0]}`;
+  }
+  return `curl ${headers} -LO ${location}`;
+};
+
+const defaultGitlabJob = ({ location, tokens }) => {
+  const filename = location.substring(location.lastIndexOf('/') + 1);
+  return `
+download:mender-tools:
+  image: curlimages/curl
+  stage: download
```

Comment:
This is a nice idea; FYI, we'll cover this (and the other CI/CD templates) in the "application updates" epic I'm currently working on. My only objection is that the gitlab job can be useful for a subset of the downloadable packages, not all of them.